### PR TITLE
Edit user public profile

### LINKED ISSUE #7: [Feature 2] Edit User Public Profile
This feature allows users to edit their public profile for GroupedUp including elements like their user bio and preferences.

### PR DESCRIPTION
**Pull request by hannahpan1**
Finalized version of the edit user public profile use case. Includes all working screens. This feature allows the user to edit their public profile which only saves if they input correct details.

### CODE REVIEW

File: src/main/java/edit_user_public_profile_screens/EditUserPublicProfileScreen.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        return meetingTime;
+    }
+
+    @Override
+    public void buildScrollPane() {
+        JScrollPane scrollPane = new JScrollPane();
+        scrollPane.setViewportView(buildProfileInfo());
+        this.add(scrollPane, BorderLayout.CENTER);
+    }
+
```
Comment (helenaglow) [inline src/main/java/edit_user_public_profile_screens/EditUserPublicProfileScreen.java:200]: Good idea making this a panel so you can put it into the scroll pane!

File: src/main/java/matching_algorithm_screens/HomeMatchesBoundary.java [hunk 2]
```diff
@@ -22,4 +23,5 @@ public interface HomeMatchesBoundary extends ListSelectionListener {
     public void setViewMyGroupsController(ViewMyGroupsController viewMyGroupsController);
 
     public void applyToGroupController(ApplyToGroupController applyToGroupController);
+    public void setViewUserProfileController(ViewUserPublicProfileController viewUserPublicProfileController);
 }
```
Comment (helenaglow) [inline src/main/java/matching_algorithm_screens/HomeMatchesBoundary.java:27]: Thank you for adding your controller to the interface

File: src/main/java/edit_user_public_profile_screens/EditUserPublicProfilePresenter.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+public class EditUserPublicProfilePresenter implements EditUserPublicProfileOutputBoundary {
+    EditUserPublicProfileScreenBoundary editUserProfileScreenBoundary;
+
+    public EditUserPublicProfilePresenter(EditUserPublicProfileScreenBoundary editUserProfileScreenBoundary) {
+        this.editUserProfileScreenBoundary = editUserProfileScreenBoundary;
+    }
+
+    /**
+     *
+     * @param EditedChanges This is the newly edited changes for the user public profile.
```
Comment (helenaglow) [inline src/main/java/edit_user_public_profile_screens/EditUserPublicProfilePresenter.java:16]: Good to see you updated to return void!

**Review by helenaglow — APPROVED** [review]
Everything looks good!

**Review by ileuxali — APPROVED** [review]

**Review by aaryavatsa — APPROVED** [review]
looking good -- are these screens connected to everyone else's?

**Pull request merged**